File: leetcode/math/1056_Confusing_Number/codes/math.c
#include <stdbool.h>

bool confusingNumber(int n) {
    int map[10];
    for (int i = 0; i < 10; ++i) map[i] = -1;
    map[0] = 0; map[1] = 1; map[6] = 9; map[8] = 8; map[9] = 6;

    int original = n, rotated = 0;
    while (n > 0) {
        int d = n % 10;
        if (map[d] == -1) return false;
        rotated = rotated * 10 + map[d];
        n /= 10;
    }
    return rotated != original;
}

File: leetcode/math/1056_Confusing_Number/codes/math.py
class Solution:
    def confusingNumber(self, n: int) -> bool:
        rotate_map = {"0":"0", "1":"1", "6":"9", "8":"8", "9":"6"}
        s = str(n)
        rotated = []
        
        for ch in reversed(s):
            if ch not in rotate_map:
                return False
            rotated.append(rotate_map[ch])
        
        return int("".join(rotated)) != n
